qiskit/ml/algorithms/quantum_kernel.py

```diff
@@ -0,0 +1,272 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Quantum Kernel Algorithm"""
+
+from typing import Optional, Union
+
+import logging
+
+import numpy as np
+
+from qiskit import QuantumCircuit, QuantumRegister, ClassicalRegister
+from qiskit.circuit import ParameterVector
+from qiskit.providers import Backend, BaseBackend
+
+from qiskit.aqua import QuantumInstance
+
+logger = logging.getLogger(__name__)
+
+
+class QuantumKernel:
+    """
+    Quantum Kernel
+
+    The general task of machine learning is to find and study patterns in data. For many
+    algorithms, the datapoints are better understood in a higher dimensional feature space,
+    through the use of a kernel function:
+
+    K(x, y) = <f(x), f(y)>.
+
+    Here K is the kernel function, x, y are n dimensional inputs. f is a map from n-dimension
+    to m-dimension space. <x,y> denotes the dot product. Usually m is much larger than n.
+
+    The quantum kernel algorithm calculates a kernel matrix, given datapoints x and y and feature
+    map f, all of n dimension. This kernel matrix can then be used in classical machine learning
+    algorithms such as support vector classification, spectral clustering or ridge regression.
+    """
+
+    BATCH_SIZE = 1000
```

Comment:
Not sure this is necessary, the `quantum_instance` splits into jobs according to the restrictions of the provider.

Reply:
I cannot recall exactly on this and the difference with what is called minibatch_size on VQC. My recollection is that some users had quite large datasets and this limit around chunking the dataset was to avoid memory issues. This is different than whatever limit might be in place around the provider (i.e. around execution via the quantum instance).

Reply:
How would I check this?

Reply:
@woodsp-ibm Thanks for the background.
@attp Can you please move `batch_size` to a constructor argument? We can set the default to `1000`. Or to `None`? I am fine with both.

Reply:
Done in c98fdbe. Decided to set the default to 1000.